### ISSUE TITLE
UI: space does not select/unselect check boxes

### ISSUE THREAD

**Issue by DaanHoogland** [apache/cloudstack#6017]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
4.16
~~~

##### SUMMARY

When (for instance?) regestering a template, a series of check-boxes can be (un)set. in 4.15 you can walk through the set with <tab>/<shift>-<tab> and select or unselect any you want <space>. In 4.16 the <space> for (un)selecting does not work anymore.

##### STEPS TO REPRODUCE

~~~
start the register new template dialog. Go to the bottom and select any of the check-boxes. Then use <tab> to go to the next or <shift>-<tab> to go to the previous and try to check/uncheck it using <space>.
~~~

##### EXPECTED RESULTS

~~~
the selection state changes
~~~

##### ACTUAL RESULTS

~~~
nothing changes
~~~

**Comment by DaanHoogland** [comment]
@utchoang I learned that this is due to the `v-ctrl-enter="handleSubmit"` that has been added to many dialog. Would you know how to enable `space` for selection on the dialogues, anyhow?

**Comment by utchoang** [comment]
@DaanHoogland I saw you create a PR to solve it. It is true that the `v-ctrl-enter` event suppressed the checkbox with a `space` event. However, for `switch`, the `space` event is not applied to it, only the `enter` event. I think there should be a similar `@keyup.space` event on `a-switch` to check/uncheck it?

**Comment by DaanHoogland** [comment]
tnx @utchoang . what do you think of #6027 ? I did it a bit brute force because I would like to have a bit uniform code. Do you think changing to the `@keyup` events makes sense? and should I revert and only apply to the registration form?

**Comment by utchoang** [comment]
@DaanHoogland I think your changes are very good. Just add `@keyup.space` event to `a-switch` to switch/unswitch is fine.

**Comment by sureshanaparti** [comment]
Fixed in #6027 for some forms, closing this issue.

@DaanHoogland @utchoang any further changes to other forms, if required, can be addressed in 4.17.  Please open another issue (with details) for tracking.
